repose-aggregator/src/docs/asciidoc/architecture/container.adoc

```diff
@@ -0,0 +1,252 @@
+              <protocol>TLSv1.2</protocol>
+          </included-protocols>
+          <tls-renegotiation-allowed>false</tls-renegotiation-allowed>
+      </ssl-configuration>
+    </deployment-config>
+</repose-container>
+----
+
+[NOTE]
+You need to specify your keystore in the container configuration just as you would in Jetty.
+
+== Diffie-Hellman Security Risk and Key Size
+Certain attacks (such as Logjam) leverage the weakness of "small" Diffie-Hellman (DH) keys.
+To mitigate the risk of such attackers, users may either exclude vulnerable ciphers, or lengthen the DH keys used by Repose.
+Instructions for the former are above.
+For the latter, note the following:
+
+"Diffie-Hellman (DH) keys of sizes less than 1024 bits have been deprecated because of their insufficient strength.
+In JDK 8, you can customize the ephemeral DH key size with the system property jdk.tls.ephemeralDHKeySize."
+
+In other words, the Java option "-Djdk.tls.ephemeralDHKeySize=2048" can be passed when starting Repose to force the use of longer DH keys.
+
+For more details, see:
+* https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html#customizing_dh_keys
+
+== SSL/TLS References
+For more information about cipher suites and which ones to dis/allow when setting up Repose, see the following references.
+
+* http://security.stackexchange.com/questions/76993/now-that-it-is-2015-what-ssl-tls-cipher-suites-should-be-used-in-a-high-securit
+* https://www.ssllabs.com/
+* https://wiki.eclipse.org/Jetty/Howto/CipherSuites#Enabling_Cipher_Suites
+* https://www.owasp.org/index.php/Transport_Layer_Protection_Cheat_Sheet
+* https://cipherli.st/
+
+== Available Ciphers and Protocols
+The list of available ciphers and protocols varies depending on the JVM.
+We have added a command line option to Repose Valve to display the available and default enabled ciphers and protocols:
+
+[source,bash]
+.Show SSL Params
+----
+java -jar /usr/share/repose/repose-valve.jar --show-ssl-params
+----
+
+This will dump a list of the default enabled SSL/TLS parameters for the JVM you're using.
+Additionally, it will list all available ciphers and protocols, should you wish to use one of those.
+
+== Running in Insecure Mode
+[WARNING]
+This mode should only be used during development testing.
+These settings are NOT intended for a production environment.
+
+When running in insecure mode, Repose will accept all certificates from external services with which it communicates (e.g. authentication service, origin service).
+
+=== Valve
+When running the Valve deployment, Repose may be placed in insecure mode by passing in the `-k` option as follows:
+[source,bash]
+.Insecure Mode - Valve
+----
+java -jar /usr/share/repose/repose-valve.jar -c /etc/repose -k
+----
+
+=== ROOT.war
+When running the ROOT.war deployment, Repose may be placed in insecure mode by passing in the following system property:
+
+[source,bash]
+.Insecure Mode - ROOT.war
+----
+-D insecure=true
```

Comment:
Having a space is normally valid. I'm not sure about this case, but I would be disappointed if it isn't.

My real concern is that this code block seems container dependent. After all, how would you do this in Glassfish?

In fact, do we know this works for Tomcat? I guess the question is, where are we adding the `-D insecure=true`? I'm pretty sure for Tomcat it would have to be set in the Tomcat startup script, a `setenv` script, or perhaps even in a web-app specific `conf`.

Of course, you could also set it as an `initParameter` in the Servlet context (defined in the `web.xml` in the WAR).